### PR TITLE
Revised error handling docs

### LINKED ISSUE #216: Improve the error handling documentation in Fusion
We currently have the [Client-side exception](https://vaadin.com/docs-beta/latest/fusion/application/client-exceptions/) article, but it’s not enough. It should be updated (and possibly renamed) to also:

- Have an introduction telling about different sources of error, such as network errors, our endpoints throwing an exception, or application code on throws on the server or on the client.
- Describe endpoint client error classes (thrown from Connect.ts):
  - `EndpointError`
    - `EndpointValidationError`
  - `EndpointResponseError`
- Provide an example in TypeScript on how to catch them and handle them with respect to the error type
  - How to extract information specific to the error
  - How to show the notification for the user about the error
- Mention the offline data fallback use case with the link to the cachable wrapper article
- Mention that the form binder is handling `EndpointValidationError` from server-side during submission

Link to error handling from the Accessing Java Backend article.

### PR DESCRIPTION
**Pull request by joheriks**
Fixes #216.

### CODE REVIEW

File: frontend/demo/fusion/errorhandling/catch-invalid-args.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import * as dataEndpoint from '../../../generated/DateEndpoint';
```
Comment (haijian-vaadin) [inline frontend/demo/fusion/errorhandling/catch-invalid-args.ts:1]: Could use the Frontend alias instead. i.e. `Frontend/generated/DateEndpoint`

File: frontend/demo/fusion/errorhandling/catch-error.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { EndpointError } from '@vaadin/flow-frontend';
+
+import * as dataEndpoint from '../../../generated/DataEndpoint';
```
Comment (haijian-vaadin) [inline frontend/demo/fusion/errorhandling/catch-error.ts:3]: Could use the Frontend alias instead. i.e. `Frontend/generated/DateEndpoint`

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]
Looks good to me in general, the imports of generated assets could be updated with the frontend alias.

File: frontend/demo/fusion/errorhandling/catch-invalid-args.ts (hunk 1 above)
Comment (joheriks) [reply frontend/demo/fusion/errorhandling/catch-invalid-args.ts]: Done.

File: frontend/demo/fusion/errorhandling/catch-error.ts (hunk 2 above)
Comment (joheriks) [reply frontend/demo/fusion/errorhandling/catch-error.ts]: Done.

**Review by haijian-vaadin — APPROVED** [review]

File: articles/fusion/application/error-handling.asciidoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+The error instance contains a field `validationErrorData` holding validation error information for each invalid parameter.
+
+For example, given the following endpoint expecting a `java.time.LocalDate` parameter:
+
+[source,Java]
+----
+include::{root}/src/main/java/com/vaadin/demo/fusion/errorhandling/DateEndpoint.java[]
+----
+
+A call with an illegal data parameter will raise an `EndpointValidationException` with information about which parameters failed validation:
```
Comment (jouni) [inline articles/fusion/application/error-handling.asciidoc:37]:
```suggestion
A call with an illegal data parameter raises an `EndpointValidationException` with information about which parameters failed validation:
```

File: articles/fusion/application/error-handling.asciidoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+If the endpoint exists and its parameters could be passed, but its execution raises a Java runtime exception, the server responds with
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500[500 Internal Server Error] and Fusion raises an error of type `EndpointError`. As a special case, if the server-side exception is an instance of `com.vaadin.flow.server.connect.exception.EndpointException` or its subclass, the server instead responds with https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400[400 Bad Request] and the exception type and message passed to the `EndpointException` in Java are available in the `EndpointError` instance via the `type` and `message` attributes.
+For example, given the following endpoint implementation:
+
+[source,Java]
+----
+include::{root}/src/main/java/com/vaadin/demo/pwa/offline/DataEndpoint.java[]
+----
+
+The following client-side call to the endpoint method will log the error message and exception type:
```
Comment (jouni) [inline articles/fusion/application/error-handling.asciidoc:57]:
```suggestion
The following client-side call to the endpoint method logs the error message and exception type:
```

File: articles/fusion/application/error-handling.asciidoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+[source,typescript]
+----
+include::{root}/frontend/demo/fusion/errorhandling/catch-error.ts[]
+----
+
+== Network Errors
+
+When the server is not reachable due to outage or network disruption, an endpoint call will result in a low-level network error, different from `EndpointError`.
```
Comment (jouni) [inline articles/fusion/application/error-handling.asciidoc:66]:
```suggestion
When the server is not reachable due to outage or network disruption, an endpoint call results in a low-level network error, different from `EndpointError`.
```

**Review by jouni — COMMENTED** [review]
Nice work 👍 

Avoiding “will” is usually easy, and recommended, so consider fixing those still.

File: articles/fusion/application/error-handling.asciidoc (hunk 3 above)
Comment (joheriks) [reply articles/fusion/application/error-handling.asciidoc]: Done.

File: articles/fusion/application/error-handling.asciidoc (hunk 4 above)
Comment (joheriks) [reply articles/fusion/application/error-handling.asciidoc]: Done.

File: articles/fusion/application/error-handling.asciidoc (hunk 5 above)
Comment (joheriks) [reply articles/fusion/application/error-handling.asciidoc]: Done.

**Comment by haijian-vaadin** [comment]
One Vale error remains, seems not introduced in this PR, but looks easy to fix.

**Pull request merged**